src/main/java/net/minecraftforge/fml/common/asm/transformers/SideTransformer.java

```diff
@@ -177,14 +180,17 @@ public void accept(MethodNode method) {
         @Override
         public void visitInvokeDynamicInsn(String name, String desc, Handle bsm, Object... bsmArgs)
         {
-            if (META_FACTORY.equals(bsm))
+            for (Object bsmArg : bsmArgs)
             {
-                Handle dynamicLambdaHandle = (Handle) bsmArgs[1];
-                dynamicLambdaHandles.add(dynamicLambdaHandle);
+                if (bsmArg instanceof Handle)
```

Comment:
Don't change this, just because a dynamic has a handle as an argument doesn't mean its a lambda. Also, this will cause duplicates due to targets having multiple handle arguments.